src/examples/hl_hdf5_inmemory_files.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c), 2022, Blue Brain Project
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#include <iostream>
+#include <vector>
+
+
+#include <highfive/H5File.hpp>
+#include <hdf5_hl.h>
+
+using namespace HighFive;
+
+class InMemoryFile: public HighFive::File {
+  public:
+    explicit InMemoryFile(std::vector<std::uint8_t> buffer)
+        : _buffer(std::move(buffer)) {
+        _hid = H5LTopen_file_image(_buffer.data(),
+                                   sizeof(_buffer[0]) * _buffer.size(),
+                                   H5LT_FILE_IMAGE_DONT_RELEASE | H5LT_FILE_IMAGE_DONT_COPY);
```

Comment:
I think you need use the returned `hid_t` by `H5Gopen(file_id, "/", H5P_DEFAULT);`, as mentioned in #648.

I'm not very familiar with libhdf5, but from my previous experiences, using `hid_t` returned by `H5LTopen_file_image()` as `_hid` won't work.

Reply:
Sorry about the delay. The answer, and part of the reason for the delay, is I simply don't know. I'm not observing the same issue, the example worked as shown. But I'm not doing anything complex in the example.

At a later point we can try and improve the example by opening and closing a group; but for the time being I hope it addresses your immediate requirements.

The one difference I'm confident does matter is the `| H5LT_FILE_IMAGE_DONT_COPY` because without it, a copy is performed and the first `*_DONT_RELEASE` is ignored.

Reply:
Thanks. Introduce the protect ctor and using a aubclass does work for me.